File: project101/summary.py

rm(list=ls())
setwd('/lustre/wangjc01/huzixin/deconv/data')

mse <- function(x,y){
  sum((x-y)^2)/sum((y-mean(y))^2)
}

minmax <- function(x){
  (x-min(x)) / (max(x)-min(x))
}

bulk_rush <- apply(read.csv('bulk.csv'),2,minmax)
bulk_adni <- apply(read.csv('bulk_adni.csv'),2,minmax)

setwd('/lustre/wangjc01/huzixin/deconv/log/rlt_adni')
fit.adni <- lapply(dir(pattern='bulk_fit_'),read.csv)
names(fit.adni) <- dir(pattern='bulk_fit_')

setwd('/lustre/wangjc01/huzixin/deconv/log/rlt_rush')
fit.rush <- lapply(dir(pattern='bulk_fit_'),read.csv)
names(fit.rush) <- dir(pattern='bulk_fit_')

cbind(sapply(fit.adni,function(x){mse(x,y=bulk_adni)}))
cbind(sapply(fit.rush,function(x){mse(x,y=bulk_rush)}))

File: project101/summary.R

rm(list=ls())
source('/Users/wenrurumon/Documents/uthealth/deconv/ae/model.R')

setwd("/Users/wenrurumon/Documents/uthealth/deconv/adni_data")
bulk <- fread('bulk_adni.csv')
ref <- fread('ref_adni.csv')

setwd("/Users/wenrurumon/Documents/uthealth/deconv/ae")
bulk <- fread('bulk.csv') 
ref <- fread('reference.csv')

x <- ref %>% as.matrix 
y <- bulk %>% as.matrix
x.sel <- apply(x,1,function(xi){
  names(which(xi>quantile(xi,0.99)))
}) %>% unlist %>% unique
y.sel <- y[,colnames(y)%in%x.sel]
x.sel <- x[,colnames(x)%in%x.sel]
dim(y.sel)
system.time(deconv.sel <- deconv(t(x.sel),t(y.sel),ifprint=T))
fit.sel <- dedeconv((x),deconv.sel$coef,(y))
mse.sel <- mse(fit.sel,y)
rlt.sel <- list(fit=fit.sel,mse=mse.sel)

system.time(deconv.sel <- deconv.lm(t(x.sel),t(y.sel),ifprint=T))
fit.sel <- dedeconv((x),deconv.sel,(y))
mse.sel <- mse(fit.sel,y)
rlt.sel <- list(fit=fit.sel,mse=mse.sel)
